### ISSUE TITLE
Firewall rules for terraform-google-network module

### ISSUE THREAD

**Issue by ZLPG23** [terraform-google-modules/terraform-google-network#461]
### TL;DR

Hi there, one of our enterprise customers currently uses terraform-google-modules/terraform-google-network in their IaC pipeline, and they would like to have source and destination to be used in egress and ingress firewall rules. 

### Additional information

just FYI: Below is the comment from our customer: 
***
Seems like it doesn’t support ingress rule with destination range, neither egress rule with source range.
 
https://github.com/terraform-google-modules/terraform-google-network/blob/v7.0.0/modules/firewall-rules/main.tf#L24
  source_ranges           = each.value.direction == "INGRESS" ? each.value.ranges : null
  destination_ranges      = each.value.direction == "EGRESS" ? each.value.ranges : null
***
